### PR TITLE
PIOS_DAC: Implement DAC driver

### PR DESCRIPTION
**Pull request by mlyle**
This is WIP because it needs actual end-user useful functionality to merge.

May want to crank sample rate up too, because stair steps are fairly bad without an anti-aliasing filter.

### CODE REVIEW

**Comment by mlyle** [comment]
Need a decent way to turn this on and off, and I think the pios_dac priority system needs a little help.

With #1525 plus this, I get:

image

**Comment by mlyle** [comment]
With #1525, this works well ---- though configuration is necessary.  An annunciator target would be really nice.

An example from seppuku is at http://jar.lyle.org/~mlyle/data.wav

A crude demodulator is at: https://gist.github.com/mlyle/4e0db1cab991cdceb364dc2859c3099e

Which yields output like:

```
595 ___m8}____=____]m_Y____c_6_'__h_)__9_G_a_~n_____o________NN
687 $TA_______
778 $TA_______
944 $TG_______________
1035 $TA____B_A
1127 $TA____B_A
1228 $TS______NN
```

(Nonprintables become underscores)

**Comment by mlyle** [comment]
@peabody124 can you easily feed that data.wav to your tools to see whether we're interoperable?

**Comment by mlyle** [comment]
envelope

http://jar.lyle.org/~mlyle/beepy.wav

I've tested both modes and selecting them from within the uavo browser.  works; I think this is ready for review.

**Comment by peabody124** [comment]
@mlyle I will try and check Sunday. On call today. Ping me on IRC then too if you don't hear anything

**Comment by mlyle** [comment]
@peabody124 Thanks, much appreciated :D

**Comment by mlyle** [comment]
Good news: it works, even before the minor magnitude adjustment above:

http://streamable.com/i3rh4

Bad news: I shorted my VTX while playing with the disassembled quad and it caught fire.  Woops.  Need to rewire stuff now.

File: flight/PiOS/STM32F4xx/inc/pios_dac.h [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+#ifndef _PIOS_DAC_H
+#define _PIOS_DAC_H
+
+#include <stdint.h>
+#include <stdio.h>
+
+#include "stm32f4xx_dac.h"
+
+typedef struct dac_dev_s *dac_dev_t;
```
Comment (tracernz) [inline flight/PiOS/STM32F4xx/inc/pios_dac.h:40]: Prefix with `pios_`?

File: flight/PiOS/inc/pios_annuncdac.h [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+ * must be maintained in each individual source file that is a derivative work
+ * of this source file; otherwise redistribution is prohibited.
+ */
+
+#ifndef _PIOS_ANNUNCDAC_H
+#define _PIOS_ANNUNCDAC_H
+
+#include <pios_dac.h>
+
+typedef struct annuncdac_dev_s *annuncdac_dev_t;
```
Comment (tracernz) [inline flight/PiOS/inc/pios_annuncdac.h:39]: `pios_`

File: flight/PiOS/inc/pios_fskdac.h [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+ * must be maintained in each individual source file that is a derivative work
+ * of this source file; otherwise redistribution is prohibited.
+ */
+
+#ifndef _PIOS_FSKDAC_H
+#define _PIOS_FSKDAC_H
+
+#include <pios_dac.h>
+
+typedef struct fskdac_dev_s *fskdac_dev_t;
```
Comment (tracernz) [inline flight/PiOS/inc/pios_fskdac.h:39]: `pios_`

**Review by tracernz — CHANGES_REQUESTED** [review]
Looks pretty good. :+1:

File: flight/PiOS/STM32F4xx/inc/pios_dac.h (hunk 1 above)
Comment (mlyle) [reply flight/PiOS/STM32F4xx/inc/pios_dac.h:40]: Last several drivers we merged like ws2811, max7456, lis3mdl we had decided not to do that and to move towards changing the convention.   Want to revisit?

Comment (tracernz) [reply flight/PiOS/STM32F4xx/inc/pios_dac.h:40]: Ah yes. I still think this kind of artificial namespacing is good but the horse has already bolted.

**Pull request merged**